### PR TITLE
Test set packet type

### PR DESCRIPTION
**Pull request by LembituNN**
Later updates to dev, now also included service_mode timeout functionality as default.

### CODE REVIEW

File: src/lib/devp_comms.c [hunk 1]
```diff
@@ -505,9 +509,16 @@ static void sleep_block_comms (comms_layer_t * p_comms)
 		{
 			if(NULL != mp_sleep_ctrl[i]) // Sleep control is optional
 			{
-				debug1("block %d", i);
+				#ifdef SERVICE_MODE_TIMEOUT
+				m_rcomm_keepalive_timeout = devp_service_mode_timeout_get();
+				#endif //SERVICE_MODE_TIMEOUT
+				if(!m_rcomm_keepalive_timeout)
```
Comment (germandevelop) [inline src/lib/devp_comms.c:515]: Can you guarantee, that 'm_rcomm_keepalive_timeout' is zero by default ?

File: src/params/devp_service_timeout.c [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+void devp_service_mode_timeout_init ()
+{
+    m_service_mode_timeout = DEVP_SERVICE_TIMEOUT_MS;
+    devp_register(&m_dp_service_mode_timeout);
+}
+
+uint32_t devp_service_mode_timeout_get ()
+{
+    return m_service_mode_timeout;
```
Comment (kobilozor) [inline src/params/devp_service_timeout.c:51]: Different indentations in the file.

File: src/lib/devp_comms.c (hunk 1 above)
Comment (LembituNN) [reply src/lib/devp_comms.c:515]: added initialization in init function.

File: src/params/devp_service_timeout.c (hunk 2 above)
Comment (LembituNN) [reply src/params/devp_service_timeout.c:51]: standardized, now tabs replaced with commas

File: src/lib/devp_comms.c [hunk 3]
```diff
@@ -505,9 +509,16 @@ static void sleep_block_comms (comms_layer_t * p_comms)
...
+				{
+					m_rcomm_keepalive_timeout = DEVP_DEFAULT_SLEEP_TIMEOUT_MS;
+				}
+				debug1("block %d for %d sec", i, m_rcomm_keepalive_timeout/1000);
 				comms_sleep_block(mp_sleep_ctrl[i]);
-				osTimerStart(m_sleep_timer, DEVP_SLEEP_TIMEOUT_MS);
+				osTimerStart(m_sleep_timer, m_rcomm_keepalive_timeout);
```
Comment (kobilozor) [inline src/lib/devp_comms.c:521]: I think if the timer is started with a tick count of 0, then it expires immediately. Is it expected behavior? Or maybe if m_rcomm_keepalive_timeout is 0, then timer should not be even started?

**Review by germandevelop — APPROVED** [review]

File: src/lib/devp_comms.c (hunk 3 above)
Comment (LembituNN) [reply src/lib/devp_comms.c:521]: timer should not be started when tick count is 0.
that's why there is check to set tick count to default if value got from devp happens to be 0.

**Review by kobilozor — APPROVED** [review]

**Pull request merged**